src/librustc_codegen_ssa/mir/operand.rs

```diff
@@ -243,14 +243,22 @@ impl<'a, 'tcx: 'a, V: CodegenObject> OperandRef<'tcx, V> {
             _ => bug!("OperandRef::extract_field({:?}): not applicable", self)
         };
 
+        let bitcast = |bx: &mut Bx, val, ty| {
+            if ty == bx.cx().type_i1() {
+                bx.trunc(val, ty)
+            } else {
+                bx.bitcast(val, ty)
+            }
+        };
+
         // HACK(eddyb) have to bitcast pointers until LLVM removes pointee types.
```

Comment:
Can you add another comment that mentions that we also need to truncate bool union fields?
Also, the comment should be above the closure.